gem/refactorise.py

```diff
@@ -211,6 +211,23 @@ def stop_at(expr):
     return result
 
 
+@_collect_monomials.register(MathFunction)
+def _collect_monomials_mathfunction(expression, self):
+    name = expression.name
+    if name in {"conj", "real", "imag"}:
+        # These are allowed to be applied to arguments, and hence must
+        # be dealt with specially. Just push the function onto each
+        # entry in the monomialsum of the child.
+        child_ms, = map(self, expression.children)
+        result = MonomialSum()
+        for k, v in child_ms.monomials.items():
+            result.monomials[k] = MathFunction(name, v)
```

Comment:
So you only propagate these complex functions to the "rest" factors, which means you assume that "atomics" are real.  This is true for argument factorisation (which is the main use of the refactoriser), but I'm not sure that this would work correctly with tensor mode (which uses the refactoriser for a different refactorisation).

I think this special assumption should at least be noted in a comment.

Reply:
`atomics` in this case (arguments) are the basis tabulations, yes? I guess the "right thing" would be to check what atomics are and propagate to both (if necessary). Or one could have a flag in the refactoriser that says "are atomics real?". WDYT?

Reply:
Added a comment.